dima/react-movies/src/App.tsx

```diff
@@ -0,0 +1,111 @@
+import * as React from "react";
+import { Component } from "react";
+import MainPage from "./pages/mainpage";
+import Movies from "./api/movies";
+import "./App.css";
+import MoviePage from "./pages/moviepage";
+import MovieInterface from "./interfaces/movieInterface";
+
+interface AppState {
+  movie: MovieInterface;
+  movies: MovieInterface[];
+  searchedMovies: null | MovieInterface[];
+  currentSortType: string;
+}
+class App extends Component<{}, AppState> {
+  state: AppState = {
+    movie: Movies[0],
+    movies: Movies,
+    searchedMovies: null,
+    currentSortType: "release date",
+  };
+  componentDidMount() {
+    this.sortMovies(this.state.currentSortType);
+  }
+  setCurrentSortType = (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ): void => {
+    const { value: currentSortType } = e.currentTarget;
+    if (currentSortType !== this.state.currentSortType) {
+      this.sortMovies(currentSortType);
+      this.setState({ currentSortType });
+    }
+  };
+  searchMovies = (value: string, category: string): MovieInterface[] => {
```

Comment:
Really nice that you had time and did a sorting and filtering 👍 . But in future (redux + redux-saga task) such things will be made on backend side, so this code will be removed.